scigym/envs/__init__.py

```diff
@@ -8,3 +8,8 @@
     id='CartPole-test-v0',
     entry_point='scigym.envs.test_envs:CartPoleEnv',
 )
+
+register(
+    id='surfacecode-v0',
+    entry_point='scigym.envs.surfacecode:SurfaceCodeEnv',
```

Comment:
Do you think the name `surfacecode` is too generic? Do you think we could move it to a folder `surfacecode_decoder`? Does that make sense? For the environment file, the name is not as important, so you may leave it as it is. In the future, we might have many environments in different folders, and maybe already having a folder named  `surfacecode` is too generic? Or would rather leave it? I don't really mind either way, just a thought.

Reply:
on the one hand I agree that surface code is perhaps too generic. It could be that other users want to contribute additional surface code type environments, perhaps with different error models, lattice sizes or rules of the game (i.e. how terminal states are decided). Additionally there could also be environments which deal with "dynamic" surface codes, as for example in lattice surgery. 

On the other hand, I think that renaming the folder won't solve the problem, in that the first type of examples I gave above there would still be a naming conflict, and the second type could easily be housed in a distinct folder called something distinct like ``surfacecode_computing". For now my preference would be to leave the name as is - as the environment really is a straightforward simulation of a surface code - and if in future someone would like to contribute another surfacecode type environment, then we can ask what makes it distinct from this one, and try incorporate that difference into the name of the folder which houses the newly contributed environment.

Reply:
What's the problem with 'surfacecode_decoding' and 
```
id=surfacecode-decoder-v0
entry_point=`scigym.envs.surfacecode_decoding:SurfaceCodeEnv
```
?
I think that it makes sense since we might want to move towards sub-subfolder, e.g.
```
qec
|----surfacecode
      |----surfacecode_decoding
      |----surfacecode_computing
      |----surfacecode_optimizing
```
In particular, my environment might appear as `surfacecode_optimizing`. And if we both name them as above, their relation is better understood. I would also not want to make this a competition about whose environment is more a `surfacecode`.  :smiley:

Reply:
Ok, I like your suggestions - i.e. `surfacecode_decoding` and `surfacecode_optimizing` make sense to me (I found `surfacecode_decoder` not quite right, as the agent is the decoder, not the environment).

With regard to #4 - I agree that more structure is better, and your suggestion there is perfect. I will introduce the hierachy now.

Reply:
Ok, the directory structure suggested in #4 has been implemented in commit  [e0b0f91](https://github.com/HendrikPN/scigym/pull/1/commits/e0b0f91708ea6e269e39a9ffffee4a26f06f0af3).

Reply:
You didn't change the id:
```
id='surfacecode-decoding-v0'
```
Otherwise, thanks for introducing the structure. This should close #4

Reply:
Ah, I have now updated the id.